android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/app/MainActivity.kt

```diff
@@ -1,23 +1,117 @@
+import com.mapbox.navigation.core.internal.extensions.flowRoutesUpdated
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.trip.session.BannerInstructionsObserver
 import com.mapbox.navigation.examples.androidauto.CarAppSyncComponent
-import com.mapbox.navigation.examples.androidauto.databinding.MapboxActivityNavigationViewBinding
+import com.mapbox.navigation.examples.androidauto.databinding.ActivityMainBinding
+import com.mapbox.navigation.examples.androidauto.databinding.LayoutDrawerMenuNavViewBinding
+import com.mapbox.navigation.examples.androidauto.utils.NavigationViewController
+import com.mapbox.navigation.examples.androidauto.utils.TestRoutes
+import com.mapbox.navigation.ui.base.installer.installComponents
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import com.mapbox.navigation.ui.maps.guidance.junction.api.MapboxJunctionApi
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.launch
+
+class MainActivity : DrawerActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var menuBinding: LayoutDrawerMenuNavViewBinding
 
-class MainActivity : AppCompatActivity() {
-    private lateinit var binding: MapboxActivityNavigationViewBinding
+    override fun onCreateContentView(): View {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        CarAppSyncComponent.getInstance().attachNavigationView(binding.navigationView)
+        return binding.root
+    }
 
+    override fun onCreateMenuView(): View {
+        menuBinding = LayoutDrawerMenuNavViewBinding.inflate(layoutInflater)
+        return menuBinding.root
+    }
+
+    private lateinit var controller: NavigationViewController
+
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        binding = MapboxActivityNavigationViewBinding.inflate(layoutInflater)
-        setContentView(binding.root)
 
-        // TODO going to expose a public api to share a replay controller
-        // This allows to simulate your location
-//        binding.navigationView.api.routeReplayEnabled(true)
+        LogConfiguration.setLoggingLevel("nav-sdk", LoggingLevel.DEBUG)
 
-        CarAppSyncComponent.getInstance().attachNavigationView(binding.navigationView)
+        controller = NavigationViewController(this, binding.navigationView)
+
+        menuBinding.toggleReplay.isChecked = binding.navigationView.api.isReplayEnabled()
+        menuBinding.toggleReplay.setOnCheckedChangeListener { _, isChecked ->
+            binding.navigationView.api.routeReplayEnabled(isChecked)
+        }
+
+        menuBinding.junctionViewTestButton.setOnClickListener {
+            lifecycleScope.launch {
+                val (or, de) = TestRoutes.valueOf(
```

Comment:
```suggestion
                val (origin, destination) = TestRoutes.valueOf(
```

nit: `or` gets confused in github since some languages it is a keyword

<img width="301" alt="Screenshot 2023-01-19 at 1 02 06 PM" src="https://user-images.githubusercontent.com/3021882/213524175-374e0d09-f5a7-4d83-a233-821e92bc1113.png">

Reply:
Done